### PR TITLE
[firebase_auth] Add missing method getIdTokenResult()

### PR DESCRIPTION
**Pull request by lukaspili**
## Description

Firebase auth is missing the method `getIdTokenResult()` which is available on native SDKs. It allows to retrieve the current token + other useful info like the token custom claims.

The PR includes the changes:

- Add new public API on dart side: `Future<IdTokenResult> getIdTokenResult(bool refresh)`
- Add new object on dart side: `IdTokenResult`
- Add new method call handler on iOS, which calls the native firebase `getIdTokenResult(BOOL refresh)`
- Update the method call handler `getIdToken` on Android, in order to handle both `getIdToken()` and `getIdTokenResult()`. The reason is the firebase Android SDK only provides one method `getIdToken`, which returns the id token result. The method now returns only the token or the whole token result, depending on the dart call.

What is missing: tests, changelog, version bump.
Will add these later.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java [hunk 1]
```diff
@@ -530,6 +531,9 @@ private void handleGetToken(MethodCall call, final Result result, FirebaseAuth f
       return;
     }
 
+    // Contrary to iOS, android has a single method to handle getIdToken() and getIdTokenResult()
```
Comment (collinjackson) [inline packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java]:
```suggestion
    // Contrary to iOS, Android has a single method to handle `getIdToken()` and `getIdTokenResult()`
```

**Comment by collinjackson** [comment]
After looking at this more carefully, I think that getIdToken() should probably return a result object that can be used to get the token. This is a breaking API change, so I'll get some internal feedback about naming.

**Comment by lukaspili** [comment]
I understand, in my opinion there should be only but 1 method. Javascript and iOS Firebase SDKs provide 2 methods, while Android SDK provides only 1.

What about deprecating `getIdToken()` (which currently returns the token string) and adding the new method `getIdTokenResult()` (which will return the result object)? 
It would allow to push without breaking change, and removing the deprecated method in the next major version.

**Comment by collinjackson** [comment]
Haven't forgotten about this one, I should have a naming decision soon and then we can move forward. Thanks for your patience.

**Comment by lukaspili** [comment]
No problem. I'll wait on your decision before implementing the tests.

**Comment by lukaspili** [comment]
Implemented changes for renaming, updated the tests and the changelog.
PTAL.

**Comment by collinjackson** [comment]
I think we should call it `authTime` rather than `authDate` to be a bit more idiomatic: this lines up with the naming conventions of the web SDK is doing, matches React Native Firebase, and to reflect that the granularity of the data includes time info. This lines up with what's happening in #1919 where I want to change `Timestamp` to `Time` in the Dart API.

**Comment by lukaspili** [comment]
Makes sense. Let me know if there is anything else I can do.

**Comment by lookfirst** [comment]
@collinjackson lol, I was just looking for this and couldn't find it in 0.11! I guess I'll have to try your new version asap. =)

**Pull request merged**